decidim-core/lib/decidim/command.rb

```diff
@@ -1,6 +1,42 @@
 # frozen_string_literal: true
 
+# This file has been copied from https://github.com/andypike/rectify/blob/master/lib/rectify/command.rb
+# We have done this so we can decouple Decidim from any Virtus dependency, which is a dead project
+# Please follow Decidim discussion to understand more https://github.com/decidim/decidim/discussions/7234
 module Decidim
-  class Command < Rectify::Command
+  class Command
+    include Wisper::Publisher
```

Comment:
Note that the failing spec is because of missing `Wisper` dependency here.

This is something I have also though that do we really need Wisper for commands? I believe the only thing we are using from Wisper is the `on` method which is mapped with the broadcast events inside the command classes.

I believe you could achieve this by implementing a simple version of the `on` method, e.g.:

```ruby
def on(event, &block)
  @events ||= {}
  @events[event] = block
end

def broadcast(event, *args)
  @events[event].call(*args) if @events[event]
end
```

This is definitely not as deep implementation as Wisper (e.g. subscribing multiple times to the same event) but I believe it might work within the Decidim context.

I never even understood exactly why the command/controller architecture was made in Decidim like this but it is another discussion. I would make it as simple as possible and this way we could still maintain backwards compatible API.

Reply:
It makes sense, but i would handle this in a separate PR, that would aim to remove the Whisper dependency.